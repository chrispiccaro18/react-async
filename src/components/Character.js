import React from 'react';
import PropTypes from 'prop-types';

function Character({ name, status, species, image }) {
  return (
    <li>
      <img src={image} alt={`image of ${name}`} />
      <h3>{name}</h3>
      <p>{`Species: ${species}, Status: ${status}`}</p>
    </li>
  );
}

Character.propTypes = {
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Character;
