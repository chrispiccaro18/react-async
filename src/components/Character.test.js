import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character comp tests', () => {
  it('creates a Character comp', () => {
    const props = {
      name: 'Rick Sanchez',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    };
    const wrapper = shallow(<Character {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
