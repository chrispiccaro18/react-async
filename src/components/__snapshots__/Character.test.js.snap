// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character comp tests creates a Character comp 1`] = `
<li>
  <img
    alt="image of Rick Sanchez"
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <h3>
    Rick Sanchez
  </h3>
  <p>
    Species: Human
  </p>
  <p>
    Status: Alive
  </p>
</li>
`;
